File: models/VertexNormal.cs
using System;
using System.Diagnostics;

namespace OSRSCache.models
{
	public class VertexNormal
	{
		public int x;
		public int y;
		public int z;
		public int magnitude;

		public virtual Vector3f normalize()
		{
			Vector3f v = new Vector3f();

			int length = (int) Math.Sqrt((double)(x * x + y * y + z * z));
			if (length == 0)
			{
				length = 1;
			}

			v.x = (float) x / length;
			v.y = (float) y / length;
			v.z = (float) z / length;

			Debug.Assert(v.x >= -1f && v.x <= 1f);
			Debug.Assert(v.y >= -1f && v.y <= 1f);
			Debug.Assert(v.z >= -1f && v.z <= 1f);

			return v;
		}
	}

}

File: src/main/java/net/runelite/cache/models/VertexNormal.cpp
#include "VertexNormal.h"

namespace net::runelite::cache::models
{

	std::shared_ptr<Vector3f> VertexNormal::normalize()
	{
		std::shared_ptr<Vector3f> v = std::make_shared<Vector3f>();

		int length = static_cast<int>(std::sqrt(static_cast<double>(x * x + y * y + z * z)));
		if (length == 0)
		{
			length = 1;
		}

		v->x = static_cast<float>(x) / length;
		v->y = static_cast<float>(y) / length;
		v->z = static_cast<float>(z) / length;

		assert(v->x >= -1.0f && v->x <= 1.0f);
		assert(v->y >= -1.0f && v->y <= 1.0f);
		assert(v->z >= -1.0f && v->z <= 1.0f);

		return v;
	}
}
